### ISSUE TITLE
Bato.to support

### ISSUE THREAD

**Issue by DakuTree** [DakuTree/manga-tracker#14]
Bato.to is the site I want to track the most, but is probably the hardest to do so.

About a year ago, bato.to moved to using an AJAX viewer. It also changed a bunch of other things which made it much harder to do any form of tracking.

Bato.to chapter URLS are formatted like "http://bato.to/reader#CHAPTERID_PAGEN". This translates via an AJAX request into "http://bato.to/areader?id=CHAPTERID&p=PAGEN". This is all pretty simple. There is a few problems here. First off, the URL gives no information to the actual chapter number, nor the title. This pretty much means we have to try and grab this manually.
The second problem, is that bato.to supports both multiple languages as well as multiple groups, which means we need to be more specific with how we track.

On the backend, we store the full title, the title URL (or rather the part) and the latest chapter. This is then parsed when needed into full chapter and title URLs.
With bato.to this isn't so easy, since we need to track multiple different elements. We can't exactly generate a URL with bato.to, or at least not easily.

One idea I had for storing chapter information, would be to simple store a delimited string containing each part needed.
For instance we could store chapter info as: "CHAPTER_ID::CHAPTER_N(TRIMMED)"
The problem here though, is we update titles by the title info, so unless that is stored there...it's a bit difficult.

Going with the idea of using DSV (Delimiter separated values), we "could" simply store the title_url as something like "TITLE_URL::::TRANSLATION_GROUP/LANG". I'm fairly sure it's possible to use cookies to force only this to show, but I'd have to look more into it.

Basically for bato.to to work completely, we need to move away from using variables for grabbing site/title/chapter info, and more using functions to grab it instead so the site can parse it accordingly.

---

I've probably overcomplicated this by a bunch, but this stuff is confusing :|
